doc/secure-boot/secure-boot-test-plan.md

```diff
@@ -0,0 +1,115 @@
+# SONiC Secure Boot Test Plan
+
+## Overview
+The purpose of this test plan is to validate that SONiC switch supports secure boot.
+
+## Test cases
+
+### 1. Test to install an signed image
+Test to install a signed image, expect the signed image can be installed.
+
+**Steps:**
+ * Copy a signed image
+ * Run the sonic installer command: sonic_installer install <image>
+ * Expect the installation successfully
+
+### 2. Test to install an unsigned image
+Test to install an unsigned image, expect the unsigned image never be installed when SONiC Secure Boot enabled.
+
+**Steps:**
+ * Create a fake unsigned image
+ * Run the sonic installer command: sonic_installer install <image>
+ * Expect the installation failed with image not signed
+
+### 3. Test fast-reboot/warm-reboot from signed image into another signed image
+
+**Steps:**
+ * Basic boot up test
+ * Allowlist not cover this scenario.
+ * Check /proc/cmdline
+ 
+### 4. Test not able to reboot from unsigned image when running fast-reboot/warm-reboot
+
+**Steps:**
+ * Replace the current image to an unsigned dummy image
+ * Run fast-reboot
+ * Expect the switch is not reboot with verification failure message
+ * Run warn-reboot
+ * Expect the switch is not reboot with verification failure message
+
+### 6. Test Boot from untrusted signed image
+
+**Steps:**
+ * Sign image using self-signed certificate not installed in switch
+ * Install the image
+ * Reboot
+ * Expect system hanging on the bootloader
+ 
+### 7. Test CA certificate expired
+Test CA rotation, CA expired need to change another CA
+
+**Steps:**
+ * Create a test CA certificate that will expired in a short time
+ * Sign the sonic image by a certificate signed by the CA
+ * Install the CA to the switch
+ * Install the sonic image
+ * Wait until the CA expired
+ * Start to boot the switch
+ * Expect system hanging on the bootloader
+
+ 
+### 8. Test tempered image
+
+**Steps:**
+ * Change a new file into the signed image
+ * Expect the reboot will be failed
+
+### 9. Test no executable files in rw folder after reboot
+If there are any files with -x option in rw folder, the option will be removed after the SONiC switch reboot.
```

Comment:
Maybe replace `-x` by `+x` ?
chmod syntax relies on + to add a flag and - to remove it.
Even though what you meant is completely understandable it might be more accurate to use +.

Reply:
Fixed.

Reply:
The line 73 is not fixed yet.

---
In reply to: [473782028](https://github.com/Azure/SONiC/pull/662#discussion_r473782028) [](ancestors = 473782028)